lib/connection/connect.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+const net = require('net');
+const tls = require('tls');
+const Connection = require('./connection');
+
+function connect(options, callback) {
+  if (options.family !== void 0) {
+    makeConnection(options.family, options, (err, socket) => {
+      if (err) {
+        return callback(err, socket); // in the error case, `socket` is the originating error event
+      }
+
+      callback(null, new Connection(socket, options));
+    });
+
+    return;
+  }
+
+  return makeConnection(6, options, (err, ipv6Socket) => {
+    if (err) {
+      // if (this.logger.isDebug()) {
```

Comment:
Why is this commented out?

Reply:
good catch, added debug calls to `connect` call site to preserve this info